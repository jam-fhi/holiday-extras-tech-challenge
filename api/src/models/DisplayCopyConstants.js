export const SERVER_COPY = {
	liveOn: 'We are live on port'
};
