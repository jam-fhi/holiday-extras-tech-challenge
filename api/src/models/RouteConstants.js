export const LOGIN = 'login';
export const BASE = 'api';
export const REGISTER = 'register';
export const UPDATE = 'update';
export const DELETE = 'delete';
export const USER = 'user';
export const ALL_USERS = 'allusers';
export const APIDOCS = 'api-docs';
