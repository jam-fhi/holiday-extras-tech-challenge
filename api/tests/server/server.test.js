import APIServer from '../../src/server';
import superagent from 'superagent';
import { PORT, HOST, LOGIN } from '../../src/models/RouteConstants';

describe('The host server will provide access to backend functionality', () => {
	const server = new APIServer();

	beforeEach(async () => {
		await server.startServer(PORT);
	});

	afterEach(async () => {
		await server.stopServer();
	});

	it('Will run the server', async () => {
		const serverReply = await superagent.get(`${HOST}:${PORT}/${LOGIN}`);
		delete serverReply.header.date;
		expect(serverReply).toMatchSnapshot();
	});
});
