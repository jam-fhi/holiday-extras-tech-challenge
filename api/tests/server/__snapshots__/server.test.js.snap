// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The host server will provide access to backend functionality Will add a new user 1`] = `
Object {
  "header": Object {
    "access-control-allow-origin": "*",
    "connection": "close",
    "content-length": "2",
    "content-type": "text/plain; charset=utf-8",
    "etag": "W/\\"2-nOO9QiTIwXgNtWtBJezz8kv3SLc\\"",
    "x-powered-by": "Express",
  },
  "req": Object {
    "data": Object {
      "about": "I like fishing",
      "email": "jerry@holextra.com",
      "familyname": "Solomon",
      "givenname": "Jerry",
      "id": 1,
      "password": "password",
      "token": "",
    },
    "headers": Object {
      "content-type": "application/json",
    },
    "method": "POST",
    "url": "http://localhost:3002/api/register",
  },
  "status": 200,
  "text": "OK",
}
`;

exports[`The host server will provide access to backend functionality Will get a user 1`] = `
Object {
  "about": "I like cars",
  "created": "2020-02-15T13:07:01.000Z",
  "email": "sally@holextra.com",
  "familyName": "Solomon",
  "givenName": "Sally",
  "id": 3,
  "password": "password",
  "token": "",
}
`;

exports[`The host server will provide access to backend functionality Will get all users 1`] = `
Object {
  "header": Object {
    "access-control-allow-origin": "*",
    "connection": "close",
    "content-length": "189",
    "content-type": "application/json; charset=utf-8",
    "etag": "W/\\"bd-Ikhq8Eb2SsAvoLE4oYy7wLzjcwI\\"",
    "x-powered-by": "Express",
  },
  "req": Object {
    "data": undefined,
    "headers": Object {},
    "method": "GET",
    "url": "http://localhost:3002/api/allusers",
  },
  "status": 200,
  "text": "[{\\"name\\":\\"Tom Solomon\\",\\"about\\":\\"I like music\\"},{\\"name\\":\\"Dick Solomon\\",\\"about\\":\\"I like films\\"},{\\"name\\":\\"Harry Solomon\\",\\"about\\":\\"I like books\\"},{\\"name\\":\\"Sally Solomon\\",\\"about\\":\\"I like cars\\"}]",
}
`;

exports[`The host server will provide access to backend functionality Will give out swagger docs 1`] = `
Object {
  "header": Object {
    "access-control-allow-origin": "*",
    "connection": "close",
    "content-length": "4181",
    "content-type": "application/json; charset=utf-8",
    "etag": "W/\\"1055-b+1dOA14K+FlNjvBLdk8xEeIatY\\"",
    "x-powered-by": "Express",
  },
  "req": Object {
    "data": undefined,
    "headers": Object {},
    "method": "GET",
    "url": "http://localhost:3002/api/api-docs",
  },
  "status": 200,
  "text": "{\\"info\\":{\\"title\\":\\"Amazing Online Profiles!\\",\\"version\\":\\"1.0.0\\",\\"description\\":\\"Technical Challenge for Holiday Extras\\"},\\"host\\":\\"localhost\\",\\"basePath\\":\\"/api\\",\\"swagger\\":\\"2.0\\",\\"paths\\":{\\"/allusers\\":{\\"get\\":{\\"description\\":\\"Gets all users in the system\\",\\"produces\\":[\\"application/json\\"],\\"consumes\\":[\\"multipart/form-data\\"],\\"responses\\":{\\"200\\":{\\"description\\":\\"all users returned\\"},\\"404\\":{\\"description\\":\\"not found error\\"},\\"500\\":{\\"description\\":\\"internal server error\\"}}}},\\"/user\\":{\\"get\\":{\\"description\\":\\"Allows a user to get their profile details\\",\\"produces\\":[\\"application/json\\"],\\"consumes\\":[\\"multipart/form-data\\"],\\"parameters\\":[{\\"name\\":\\"_id\\",\\"description\\":\\"The mongo database id for the user\\",\\"in\\":\\"formData\\",\\"required\\":true,\\"type\\":\\"string\\"}],\\"responses\\":{\\"200\\":{\\"description\\":\\"users details returned\\"},\\"404\\":{\\"description\\":\\"not found error\\"},\\"500\\":{\\"description\\":\\"internal server error\\"}}}},\\"/delete\\":{\\"delete\\":{\\"description\\":\\"Allows a user to update their profile\\",\\"produces\\":[\\"application/json\\"],\\"consumes\\":[\\"multipart/form-data\\"],\\"parameters\\":[{\\"name\\":\\"_id\\",\\"description\\":\\"The mongo database id for the user\\",\\"in\\":\\"formData\\",\\"required\\":true,\\"type\\":\\"string\\"}],\\"responses\\":{\\"200\\":{\\"description\\":\\"deleted account\\"},\\"500\\":{\\"description\\":\\"internal server error\\"}}}},\\"/update\\":{\\"patch\\":{\\"description\\":\\"Allows a user to update their profile\\",\\"produces\\":[\\"application/json\\"],\\"consumes\\":[\\"multipart/form-data\\"],\\"parameters\\":[{\\"name\\":\\"_id\\",\\"description\\":\\"The mongo database id for the user\\",\\"in\\":\\"formData\\",\\"required\\":true,\\"type\\":\\"string\\"},{\\"name\\":\\"id\\",\\"description\\":\\"It was in the requirements, but mongo uses _id. It'll be a special number.\\",\\"in\\":\\"formData\\",\\"required\\":true,\\"type\\":\\"integer\\"},{\\"name\\":\\"email\\",\\"description\\":\\"Email address to use for login.\\",\\"in\\":\\"formData\\",\\"required\\":true,\\"type\\":\\"string\\"},{\\"name\\":\\"givenname\\",\\"description\\":\\"User's given name\\",\\"in\\":\\"formData\\",\\"required\\":true,\\"type\\":\\"string\\"},{\\"name\\":\\"familyname\\",\\"description\\":\\"User's family name\\",\\"in\\":\\"formData\\",\\"required\\":true,\\"type\\":\\"string\\"},{\\"name\\":\\"password\\",\\"description\\":\\"User's password.\\",\\"in\\":\\"formData\\",\\"required\\":true,\\"type\\":\\"string\\"},{\\"name\\":\\"about\\",\\"description\\":\\"Information about a user\\",\\"in\\":\\"formData\\",\\"required\\":true,\\"type\\":\\"string\\"}],\\"responses\\":{\\"200\\":{\\"description\\":\\"updated account\\"},\\"400\\":{\\"description\\":\\"details failed validation\\"},\\"500\\":{\\"description\\":\\"internal server error\\"}}}},\\"/register\\":{\\"post\\":{\\"description\\":\\"Allows a user to register for an awesome profile\\",\\"produces\\":[\\"application/json\\"],\\"consumes\\":[\\"multipart/form-data\\"],\\"parameters\\":[{\\"name\\":\\"id\\",\\"description\\":\\"It was in the requirements, but mongo uses _id. It'll be a special number.\\",\\"in\\":\\"formData\\",\\"required\\":true,\\"type\\":\\"integer\\"},{\\"name\\":\\"email\\",\\"description\\":\\"Email address to use for login.\\",\\"in\\":\\"formData\\",\\"required\\":true,\\"type\\":\\"string\\"},{\\"name\\":\\"givenname\\",\\"description\\":\\"User's given name\\",\\"in\\":\\"formData\\",\\"required\\":true,\\"type\\":\\"string\\"},{\\"name\\":\\"familyname\\",\\"description\\":\\"User's family name\\",\\"in\\":\\"formData\\",\\"required\\":true,\\"type\\":\\"string\\"},{\\"name\\":\\"password\\",\\"description\\":\\"User's password.\\",\\"in\\":\\"formData\\",\\"required\\":true,\\"type\\":\\"string\\"},{\\"name\\":\\"about\\",\\"description\\":\\"Information about a user\\",\\"in\\":\\"formData\\",\\"required\\":true,\\"type\\":\\"string\\"}],\\"responses\\":{\\"200\\":{\\"description\\":\\"created account\\"},\\"400\\":{\\"description\\":\\"details failed validation\\"},\\"500\\":{\\"description\\":\\"internal server error\\"}}}},\\"/login\\":{\\"post\\":{\\"description\\":\\"Login to the application\\",\\"produces\\":[\\"application/json\\"],\\"consumes\\":[\\"multipart/form-data\\"],\\"parameters\\":[{\\"name\\":\\"email\\",\\"description\\":\\"Email address to use for login.\\",\\"in\\":\\"formData\\",\\"required\\":true,\\"type\\":\\"string\\"},{\\"name\\":\\"password\\",\\"description\\":\\"User's password.\\",\\"in\\":\\"formData\\",\\"required\\":true,\\"type\\":\\"string\\"}],\\"responses\\":{\\"200\\":{\\"description\\":\\"login\\"},\\"400\\":{\\"description\\":\\"login details failed validation\\"},\\"401\\":{\\"description\\":\\"login details failed authentication\\"},\\"500\\":{\\"description\\":\\"internal server error\\"}}}},\\"/api-docs\\":{\\"get\\":{\\"description\\":\\"Displays api docs\\",\\"produces\\":[\\"application/json\\"],\\"consumes\\":[\\"multipart/form-data\\"],\\"responses\\":{\\"200\\":{\\"description\\":\\"Display api docs\\"}}}}},\\"definitions\\":{},\\"responses\\":{},\\"parameters\\":{},\\"securityDefinitions\\":{},\\"tags\\":[]}",
}
`;

exports[`The host server will provide access to backend functionality Will login successfully 1`] = `
Object {
  "header": Object {
    "access-control-allow-origin": "*",
    "connection": "close",
    "content-length": "185",
    "content-type": "application/json; charset=utf-8",
    "x-powered-by": "Express",
  },
  "req": Object {
    "data": undefined,
    "headers": Object {
      "email": "sally@holextra.com",
      "password": "password",
    },
    "method": "POST",
    "url": "http://localhost:3002/api/login",
  },
  "status": 200,
  "text": undefined,
}
`;

exports[`The host server will provide access to backend functionality Will update a user 1`] = `
Object {
  "about": "I like cars",
  "email": "jerry@holextra.com",
  "familyname": "Solomon",
  "givenname": "Sally",
  "id": 3,
  "password": "password",
}
`;
